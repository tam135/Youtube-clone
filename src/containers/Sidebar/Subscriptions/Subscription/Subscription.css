/* No CSS *//*# sourceMappingURL=Subscription.css.map */