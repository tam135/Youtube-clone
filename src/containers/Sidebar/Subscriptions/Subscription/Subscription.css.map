{
    "version": 3,
    "mappings": "",
    "sources": [
        "Subscription.scss"
    ],
    "names": [],
    "file": "Subscription.css"
}