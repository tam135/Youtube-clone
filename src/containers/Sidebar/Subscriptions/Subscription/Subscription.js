import React from 'react'
import { Icon, Image, Menu } from 'semantic-ui-react';
import './Subscription.scss';


export default function Subscription() {
    
    let rightElement = null;
    const { broadcasting, amountNewVideos } = props;
    if(broadcasting) {
        rightElement = <Icon name='signal'/>;
    } else if (amountNewVideos) {
        rightElement = <span className='new-videos-count'>{amountNewVideos}</span>
    }



    return (
        <Menu.item>

        </Menu.item>
    )
}
