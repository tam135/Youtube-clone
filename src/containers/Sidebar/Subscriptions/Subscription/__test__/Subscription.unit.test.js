import React from "react";
import { shallow } from "enzyme";
import Subscription from '../Subscription';

describe('Subscription', () => {
    test('renders empty subscription', () => {
        const wrapper = shallow(
            <Subscription/>
        );
        expect(wrapper).toMatchSnapshot();
    })

    test('renders broadcasting subscription', () => {
        const wrapper = shallow(
            <Subscription />
        );
        expect(wrapper).toMatchSnapshot()
    })

    test('renders no broadcastin subscription with new videos', () => {
        const wrapper = shallow(
            <Subscription amountNewVideos={4} label='PewdiePie'/>
        )
        expect(wrapper).toMatchSnapshot()
    })
})