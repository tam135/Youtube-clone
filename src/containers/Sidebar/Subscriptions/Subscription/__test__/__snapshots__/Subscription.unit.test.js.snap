// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Subscription renders broadcasting subscription 1`] = `
<MenuItem>
  <div
    className="subscription"
  >
    <div>
      <Image
        as="img"
        avatar={true}
        src="http://via.placeholder.com/28x28"
        ui={true}
      />
      <span />
    </div>
  </div>
</MenuItem>
`;

exports[`Subscription renders empty subscription 1`] = `
<MenuItem>
  <div
    className="subscription"
  >
    <div>
      <Image
        as="img"
        avatar={true}
        src="http://via.placeholder.com/28x28"
        ui={true}
      />
      <span />
    </div>
  </div>
</MenuItem>
`;

exports[`Subscription renders no broadcastin subscription with new videos 1`] = `
<MenuItem>
  <div
    className="subscription"
  >
    <div>
      <Image
        as="img"
        avatar={true}
        src="http://via.placeholder.com/28x28"
        ui={true}
      />
      <span>
        PewdiePie
      </span>
    </div>
    <span
      className="new-videos-count"
    >
      4
    </span>
  </div>
</MenuItem>
`;
